src/test/java/org/junit/internal/runners/statements/ExpectExceptionTest.java

```diff
@@ -0,0 +1,29 @@
+package org.junit.internal.runners.statements;
+
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.runners.model.Statement;
+
+import static org.junit.Assert.*;
+
+public class ExpectExceptionTest {
+
+    @Test
+    public void givenWeAreExpectingAssumptionViolatedExceptionAndAStatementThrowingAssumptionViolatedException() {
+        ExpectException sut = new ExpectException(new StatementThrowingAssumptionViolatedException(), AssumptionViolatedException.class);
```

Comment:
Sure, having both kinds of tests is fine. The integration test is necessary to verify the bug is fixed; the unit test verifies the contract of the `ExpectException` class.

If we keep this test class, please have the class Javadoc say where the integration tests are (using`@link`)

A few suggested  changes:

1. Could you update this test to use https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/internal/runners/statements/Fail.java ? That way, you can see what exception the `Statement` is throwing by looking at the test method.
2. Extract a local variable for the expression you pass to the `ExpectException` constructor and call that variable `delegate`. 
3. I don't think we use `sut` as a variable name very often. If we don't, I suggest  naming it `expectException` so future readers wont be confused by the name.